packages/oui-file/README.md

```diff
@@ -0,0 +1,117 @@
+# File
+
+<component-status cx-design="complete" ux="rc"></component-status>
+
+## Selector
+
+### Basic
+
+```html:preview
+<oui-file model="$ctrl.modelBasic"></oui-file>
+```
+
+### Placeholder
+
+```html:preview
+<oui-file model="$ctrl.modelPlaceholder" placeholder="Placeholder example"></oui-file>
+```
+
+### Disabled
+
+```html:preview
+<oui-file model="$ctrl.modelDisabled" disabled></oui-file>
+```
+
+### File restriction
+
+<oui-message type="info" dismissable="false">`maxsize` support form validators</oui-message>
+
+```html:preview
+<form novalidate name="fileForm">
+    <oui-field label="Upload file" help-text="image/jpeg, image/png, image/gif (max size 150 KB)" size="xl">
+        <oui-file name="fileUpload" model="$ctrl.modelField" 
+            accept="image/jpeg,image/png,image/gif" 
+            maxsize="150000" 
+            required>
+        </oui-file>
+    </oui-field>
+    <p><strong><code class="oui-doc-codespan">fileForm.$valid</code> value:</strong> {{fileForm.$valid | json}}</p>
+    <p><strong><code class="oui-doc-codespan">fileForm.$submitted</code> value:</strong> {{fileForm.$submitted | json}}</p>
+    <p><strong><code class="oui-doc-codespan">fileForm["fileUpload"].$error</code> value:</strong> {{fileForm["fileUpload"].$error | json}}</p>
+    <oui-button type="submit">Submit</oui-button>
+</form>
+```
+
+## Multiple files
+
+### Basic
+
+```html:preview
+<oui-file model="$ctrl.modelMultiple" maxsize="150000" multiple></oui-file>
+```
+
+### With preview
+
+<oui-message type="info" dismissable="false">Preview works only with `image/*` files.</oui-message>
+
+```html:preview
+<oui-file model="$ctrl.modelMultiplePreview" maxsize="150000" multiple preview></oui-file>
+```
+
+## Drag & Drop area
+
+### Basic
+
+```html:preview
+<oui-file model="$ctrl.modelDroparea" maxsize="150000" droparea></oui-file>
+```
+
+### With preview
+
+<oui-message type="info" dismissable="false">Preview works only with `image/*` files.</oui-message>
+
+```html:preview
+<oui-file model="$ctrl.modelDropareaPreview" maxsize="150000" droparea preview></oui-file>
+```
+
+## API
+
+| Attribute         | Type              | Binding   | One-time binding  | Values            | Default   | Description
+| ----              | ----              | ----      | ----              | ----              | ----      | ----
+| `model`           | array&lt;file&gt; | =         | no                | n/a               | n/a       | model bound to component
```

Comment:
What's a file ? Binary data ? Doesn't it have properties ?

Reply:
https://developer.mozilla.org/fr/docs/Web/API/File